src/util/serializing/ObjectInputStream.cpp

```diff
@@ -3,27 +3,30 @@
 #include "Serializeable.h"
 #include "i18n.h"
 
-ObjectInputStream::ObjectInputStream() = default;
-
-ObjectInputStream::~ObjectInputStream() {
-    if (this->str) {
-        g_string_free(this->str, true);
-        this->str = nullptr;
+template <typename T>
+T readTypeFromSStream(std::istringstream& istream, std::string typeName) {
```

Comment:
I think adding the typeName to the exception is unnecessary. With the file and line it is quite obvious what should have been read.